lib/spack/spack/cmd/common/env_utility.py

```diff
@@ -41,14 +41,14 @@ def setup_parser(subparser):
 
 
 class AreDepsInstalledVisitor:
-    def __init__(self, context="build"):
-        if context not in ("build", "test"):
-            raise ValueError("context can only be build or test")
-
-        if context == "build":
-            self.direct_deps = ("build", "link", "run")
+    def __init__(self, context: Context = Context.BUILD):
+        if context == Context.BUILD:
+            # TODO: run deps shouldn't be required for build env.
```

Comment:
Does this mean we get failures if run is removed here?

Reply:
Yeah, this is used in `spack build-env <spec>`. If runtime deps of the root are not installed, Spack errors.

If I remember correctly this mirrors the installer: `spack install --only=package <spec>` would also error if the root has missing run deps.